File: SPOJ/HANGOVER.scala
import scala.io.Source

object Main extends App{
	val index:Array[Double] = Stream.from(2)
			.scanLeft(0.0){(v,i)=>v+(1/i.toDouble)}
			.take(300).toArray

        Source.fromInputStream(System.in).getLines().takeWhile(_!="0.00")
	.map{ d =>
		(-1-java.util.Arrays.binarySearch(index,d.toDouble))+" cards(s)"
	}.foreach(println)
}
Main.main(Array())

File: SPOJ/java/HANGOVER.java
import java.util.*;

class Main{

	public static void main(String[] args){
		float[] index = new float[300];
		for(int i=1;i<300;i++)
			index[i] = index[i-1]+(1.0f/(1+i));

		Scanner in = new Scanner(System.in);
		while(true){
			float d = in.nextFloat();
			if(d==0) break;
			int di = Arrays.binarySearch(index,d);
			if(di<0) di = -1-di;
			else di = di+1;
			System.out.println(di+" cards(s)");
		}
		in.close();
	}
}
